components/Pages/Communities/TracksAdminPage.tsx

```diff
@@ -0,0 +1,711 @@
+"use client";
+import { useState, useEffect, Fragment } from "react";
+import { useAuthStore } from "@/store/auth";
+import { useSigner } from "@/utilities/eas-wagmi-utils";
+import { MESSAGES } from "@/utilities/messages";
+import { isCommunityAdminOf } from "@/utilities/sdk/communities/isCommunityAdmin";
+import type { ICommunityResponse } from "@show-karma/karma-gap-sdk/core/class/karma-indexer/api/types";
+import { useAccount } from "wagmi";
+import { errorManager } from "@/components/Utilities/errorManager";
+import { Button } from "@/components/Utilities/Button";
+import { Spinner } from "@/components/Utilities/Spinner";
+import { PlusIcon, PencilIcon, ArchiveBoxIcon, TagIcon, XMarkIcon } from "@heroicons/react/24/outline";
+import fetchData from "@/utilities/fetchData";
+import toast from "react-hot-toast";
+import { cn } from "@/utilities/tailwind";
+import { envVars } from "@/utilities/enviromentVars";
+import { Dialog, Transition } from "@headlessui/react";
+
+interface Track {
+  id: string;
+  name: string;
+  description?: string;
+  communityUID: string;
+  isArchived: boolean;
+  createdAt: Date;
+  updatedAt: Date;
+  programId?: string;
+  isActive?: boolean;
+}
+
+const TRACK_API = {
+  TRACKS: {
+    ALL: (communityUID: string, includeArchived: boolean = false) => 
+      `/tracks?communityUID=${communityUID}${includeArchived ? '&includeArchived=true' : ''}`,
+    BY_ID: (id: string) => `/tracks/${id}`,
+    CREATE: () => `/tracks`,
+    UPDATE: (id: string) => `/tracks/${id}`,
+    ARCHIVE: (id: string) => `/tracks/${id}`,
+  },
+  PROGRAMS: {
+    TRACKS: {
+      ALL: (programId: string, chainID: number) => `/programs/${programId}/tracks?chainID=${chainID}`,
+      ASSIGN: (programId: string, chainID: number) => `/programs/${programId}/tracks?chainID=${chainID}`,
+      REMOVE: (programId: string, chainID: number, trackId: string) => 
+        `/programs/${programId}/tracks/${trackId}?chainID=${chainID}`,
+    }
+  }
+};
```

Comment:
This should be called from and implemented in helper constant `INDEXER`, where we store all endpoints to be easier to replace if something change in endpoint name.